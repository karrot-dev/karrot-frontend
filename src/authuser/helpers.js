import { unref } from "vue";

import { useAuthService } from "@/authuser/services";

export function useAuthHelpers() {
  const { userId: currentUserId } = useAuthService();

  return {
    getIsCurrentUser,
  };
}
