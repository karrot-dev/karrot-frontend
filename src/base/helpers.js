import { computed } from 'vue'

import { useConfigQuery } from '@/base/queries'

export function useConfig (key) {
  const { config } = useConfigQuery()
  return computed(() => getIn(key, config.value))
}

function getIn (path, obj) {
  return path.split('.').reduce((o, key) => o?.[key], obj)
}
