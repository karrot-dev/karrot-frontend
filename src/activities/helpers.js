import { useAuthService } from '@/authuser/services'
import i18n from '@/base/i18n'
import reactiveNow from '@/utils/reactiveNow'

export function useActivityHelpers () {
  const { userId } = useAuthService()

  function getIsUserMember (activity) {
    return activity.participants.includes(userId.value)
  }

  function getIsEmpty (activity) {
    return activity.participants.length === 0
  }

  function getIsFull (activity) {
    return activity.maxParticipants > 0 && activity.participants.length >= activity.maxParticipants
  }

  function getHasStarted (activity) {
    return activity.date <= reactiveNow.value && activity.dateEnd > reactiveNow.value
  }

  function getIsUpcoming (activity) {
    return activity.date > reactiveNow.value
  }

  function getIsStartedOrUpcoming (activity) {
    return activity.dateEnd > reactiveNow.value
  }

  return {
    getIsUserMember,
    getIsEmpty,
    getIsFull,
    getHasStarted,
    getIsUpcoming,
    getIsStartedOrUpcoming,
  }
}

export function useActivityTypeHelpers () {
  function getColorName (activityType) {
    if (!activityType) return
    const { id } = activityType
    return `activity-type-${id}`
  }

  function getTranslatedName (activityType) {
    if (!activityType) return
    const { name, nameIsTranslatable } = activityType
    return nameIsTranslatable ? i18n.t(`ACTIVITY_TYPE_NAMES.${name}`) : name
  }

  function getIconProps (activityType) {
    if (!activityType) return
    const { icon } = activityType
    const title = getTranslatedName(activityType)
    return {
      name: icon,
      color: getColorName(activityType),
      title,

      // can use this in QFabAction if we set "icon" and "label" props...
      // QIcon seems to be OK with these being set too...
      // TODO: make sure it's OK to use with QIcon, otherwise create a getFabActionProps function...
      icon,
      label: title,
    }
  }

  function getFeedbackIconProps (activityType) {
    if (!activityType) return
    const { feedbackIcon } = activityType
    return {
      ...getIconProps(activityType),
      name: feedbackIcon,
    }
  }

  return {
    getColorName,
    getTranslatedName,
    getIconProps,
    getFeedbackIconProps,
  }
}
