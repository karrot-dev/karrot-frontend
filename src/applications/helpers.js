import { useAuthService } from '@/authuser/services'
import { useApplicationListQuery } from '@/applications/queries'

export function useApplicationHelpers () {
  const { userId } = useAuthService()
  const { applications } = useApplicationListQuery({ userId, status: 'pending' })

  function getHasMyApplicationPending (groupId) {
    if (!groupId) return false
    return applications.value.find(application => application.group === groupId)
  }

  return {
    getHasMyApplicationPending,
  }
}
