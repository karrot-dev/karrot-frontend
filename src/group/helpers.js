import { unref } from 'vue'

import { useAuthService } from '@/authuser/services'
import { useCurrentGroupService } from '@/group/services'

export function useGroupHelpers () {
  const { groupId: currentGroupId } = useCurrentGroupService()

  function getIsCurrentGroup (group) {
    return currentGroupId.value === unref(group).id
  }

  return {
    getIsCurrentGroup,
  }
}

export function useMembershipHelpers () {
  const {
    userId: authUserId,
  } = useAuthService()

  const {
    group,
  } = useCurrentGroupService()

  function getIsEditor (membership) {
    return membership.roles.includes('editor')
  }

  function getIsTrusted (membership) {
    return membership.trustedBy.includes(authUserId.value)
  }

  function getTrustProgress (membership) {
    return getIsEditor(membership) ? 1 : membership.trustedBy.length / group.value.trustThresholdForNewcomer
  }

  return {
    getIsEditor,
    getIsTrusted,
    getTrustProgress,
  }
}

export function useCurrentGroupId () {
  const { groupId } = useCurrentGroupService()
  return groupId
}

export function useIsEditor () {
  const { isEditor } = useCurrentGroupService()
  return isEditor
}
