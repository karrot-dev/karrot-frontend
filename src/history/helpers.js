import { useI18n } from 'vue-i18n'

import { useActivityTypeHelpers } from '@/activities/helpers'
import { useActivityTypeService } from '@/activities/services'
import { usePlaceService } from '@/places/services'

export function useHistoryHelpers () {
  const { t } = useI18n()

  const { getPlaceById } = usePlaceService()
  const { getActivityTypeById } = useActivityTypeService()

  const { getTranslatedName } = useActivityTypeHelpers()

  function getHistoryDescription (history) {
    const place = getPlaceById(history.place)

    const msgValues = {}
    if (place) {
      Object.assign(msgValues, {
        placeName: place.name,
        storeName: place.name,
        name: place.name,
      })
    }
    if (history.payload && history.payload.activityType) {
      const activityType = getActivityTypeById(history.payload.activityType)
      Object.assign(msgValues, {
        activityType: getTranslatedName(activityType),
      })
    }
    else if ([
      'ACTIVITY_TYPE_CREATE',
      'ACTIVITY_TYPE_MODIFY',
    ].includes(history.typus) && history.after) {
      const { name, nameIsTranslatable } = history.after
      Object.assign(msgValues, {
        activityType: nameIsTranslatable ? t(`ACTIVITY_TYPE_NAMES.${name}`) : name,
      })
    }
    else {
      // Generic name incase the payload doesn't not provide activityType
      Object.assign(msgValues, {
        activityType: t('GROUP.ACTIVITY'),
      })
    }
    if (history.typus === 'APPLICATION_DECLINED') {
      Object.assign(msgValues, {
        applicantName: history.payload.applicantName,
      })
    }

    return t(`HISTORY.${history.typus}`, msgValues)
  }

  return {
    getHistoryDescription,
  }
}
