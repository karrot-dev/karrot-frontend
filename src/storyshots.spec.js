// Load full build instead of runtime-only to have the compiler available
// jest.mock('vue', () => require('vue/dist/vue.common.js'))

/** Storybook has some unwanted side effects and we actually don't need it to test the stories
 * Therefore, we mimick the Storybook API to get the components and then run the snapshot tests
*/
let mockStories = []
jest.mock('@storybook/vue', () => ({
  storiesOf: (kind) => {
    const api = { kind }
    const kindStories = { kind, stories: [] }
    mockStories.push(kindStories)
    api.add = (storyName, getStory) => {
      kindStories.stories.push({
        name: storyName,
        render: getStory,
      })
      return api
    }
    return api
  },
}))

import glob from 'glob'
import lolex from 'lolex'
import { createRenderer } from 'vue-server-renderer'
import { polyfillRequestAnimationFrame, mountWithDefaults } from '>/helpers'
polyfillRequestAnimationFrame()

// To get properly faked dates, install fake Date object before importing stories
const now = new Date('2017-12-24T12:00:00Z')
const clock = lolex.install({ now, toFake: ['Date'] })

afterAll(() => {
  clock.uninstall()
})

// Mock RandomArt because it doesn't play nicely with JSDOM
jest.mock('@/components/General/RandomArt', () => ({
  functional: true,
  render: (h, context) => h('div', context.data, context.children),
}))

// Mock locales
jest.mock('@/locales', () => ({
  de: {
    name: 'Deutsch',
    locale: 'de',
    messages: () => import('@/locales/locale-de.json'),
    dateFnsLocale: () => import('date-fns/locale/de'),
  },

  en: {
    name: 'English',
    locale: 'en',
    messages: () => import('@/locales/locale-en.json'),
    dateFnsLocale: () => import('date-fns/locale/en'),
  },
}))

// Mock translation status
jest.mock('@/locales/translationStatus.json', () => ({
  'de': '13',
  'en': '42',
}))

// Mock annoying components (e.g. too much vuex dependency)
jest.mock('@/components/Settings/VerificationWarning', () => {})

const files = glob.sync('**/*.story.js', { absolute: true })
for (const f of files) {
  require(f)
}

for (const group of mockStories) {
  describe('Storyshots', () => {
    describe(group.kind, () => {
      for (const story of group.stories) {
        it(story.name, async () => {
          jest.resetModules()

          // get the component from storybook
          const component = story.render()

          // hack: translations don't work if i18n is in component, so delete it
          delete component.i18n

          const wrapper = mountWithDefaults(component)

          // use server side renderer to get renderered html string
          const renderer = createRenderer()
          expect(renderer.renderToString(wrapper.vm)).resolves.toMatchSnapshot()
        })
      }
    })
  })
}
