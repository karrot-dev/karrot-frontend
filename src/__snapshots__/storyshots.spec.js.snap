// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots activities/components/ActivityEdit Normal 1`] = `
<div class="edit-box">
  <div class="q-mb-lg">
    <!--v-if-->
  </div>
  <form class="q-gutter-y-lg" style="max-width: 700px;">
    <!--v-if-->
    <div class="row q-mt-xs"><label class="q-field row no-wrap items-start q-field--standard q-input q-mr-sm" for="f_01010101-0101-4101-8101-010101010101">
        <div class="q-field__before q-field__marginal row no-wrap items-center"><i class="q-icon notranslate material-icons" aria-hidden="true" role="presentation">access_time</i></div>
        <div class="q-field__inner relative-position col self-stretch">
          <div class="q-field__control relative-position row no-wrap" tabindex="-1">
            <div class="q-field__control-container col relative-position row no-wrap q-anchor--skip"><input class="q-field__native q-placeholder" tabindex="0" size="9" id="f_01010101-0101-4101-8101-010101010101" type="text">
              <!---->
            </div>
          </div>
          <!---->
        </div>
        <!---->
      </label><label class="q-field row no-wrap items-start q-field--standard q-input" for="f_01010101-0101-4101-8101-010101010101">
        <!---->
        <div class="q-field__inner relative-position col self-stretch">
          <div class="q-field__control relative-position row no-wrap" tabindex="-1">
            <div class="q-field__control-container col relative-position row no-wrap q-anchor--skip"><input class="q-field__native q-placeholder" tabindex="0" size="3" id="f_01010101-0101-4101-8101-010101010101" type="text">
              <!---->
            </div>
          </div>
          <!---->
        </div>
        <div class="q-field__after q-field__marginal row no-wrap items-center"><button class="q-btn q-btn-item non-selectable no-outline q-btn--flat q-btn--round q-btn--actionable q-focusable q-hoverable q-ml-xs" style="font-size: 8px;" tabindex="0" type="button"><span class="q-focus-helper"></span><span class="q-btn__content text-center col items-center q-anchor--skip justify-center row"><i class="q-icon fas fa-plus" aria-hidden="true" role="img"> </i></span></button></div>
      </label>
      <!--v-if-->
      <div class="q-ml-lg col-12 q-field__bottom">
        <div>When should it take place?</div>
      </div>
    </div>
    <div><label class="q-field row no-wrap items-start q-field--standard q-input q-field--float q-field--labeled q-field--with-bottom" for="f_01010101-0101-4101-8101-010101010101">
        <div class="q-field__before q-field__marginal row no-wrap items-center"><i class="q-icon notranslate material-icons" aria-hidden="true" role="presentation">group</i></div>
        <div class="q-field__inner relative-position col self-stretch">
          <div class="q-field__control relative-position row no-wrap" tabindex="-1">
            <div class="q-field__control-container col relative-position row no-wrap q-anchor--skip"><input class="q-field__native q-placeholder" style="max-width: 100px;" tabindex="0" aria-label="Max Slots" modelmodifiers="[object Object]" placeholder="unlimited" id="f_01010101-0101-4101-8101-010101010101" type="number">
              <div class="q-field__label no-pointer-events absolute ellipsis">Max Slots</div>
              <div class="q-slider q-slider--h q-slider--inactive inline no-wrap column q-slider--enabled q-slider--editable q-slider--label q-mx-sm self-end" role="slider" aria-valuemin="1" aria-valuemax="10" aria-orientation="horizontal" data-step="1" aria-valuenow="2" style="min-width: 60px;">
                <div class="q-slider__track-container q-slider__track-container--h" tabindex="0">
                  <div class="q-slider__track relative-position no-outline" style="height: 4px;">
                    <div class="q-slider__inner absolute" style="left: 0%; width: 100%;"></div>
                    <div class="q-slider__selection absolute" style="left: 0%; width: 11.11111111111111%;"></div>
                    <div class="q-slider__markers q-slider__markers--h absolute overflow-hidden" style="left: 0%; width: 100%; background-size: 11.11111111111111% 2px;"></div>
                    <div class="q-slider__thumb q-slider__thumb--h q-slider__thumb--h-ltr absolute non-selectable" style="width: 20px; height: 20px; left: 11.11111111111111%;"><svg class="q-slider__thumb-shape absolute-full" viewBox="0 0 20 20" aria-hidden="true">
                        <path d="M 4, 10 a 6,6 0 1,0 12,0 a 6,6 0 1,0 -12,0"></path>
                      </svg>
                      <div class="q-slider__focus-ring fit"></div>
                      <div class="q-slider__pin q-slider__pin--h q-slider__pin--h-standard absolute fit no-pointer-events">
                        <div class="q-slider__label q-slider__label--h q-slider__label--h-standard" style="min-width: 20px;">
                          <div class="q-slider__text-container q-slider__text-container--h q-slider__text-container--h-standard" style="transform: translateX(calc(-0.7777777777777778 * 20px / 2 + 38.888888888888886%));"><span class="q-slider__text">2</span></div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class="q-field__bottom row items-start q-field__bottom--animated">
            <transition-stub>
              <div class="q-field__messages col">
                <div>How many people should participate?</div>
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <div class="q-field__after q-field__marginal row no-wrap items-center">
          <!--v-if-->
        </div>
      </label>
      <!--v-if-->
    </div>
    <div>
      <div class="mentionable" style="position: relative;">
        <div class="relative-position q-mb-sm"><label class="q-field row no-wrap items-start q-field--standard q-textarea q-textarea--autogrow q-field--float q-field--labeled q-field--with-bottom" for="f_01010101-0101-4101-8101-010101010101">
            <div class="q-field__before q-field__marginal row no-wrap items-center"><i class="q-icon notranslate material-icons" aria-hidden="true" role="presentation">info</i></div>
            <div class="q-field__inner relative-position col self-stretch">
              <div class="q-field__control relative-position row no-wrap" tabindex="-1">
                <div class="q-field__control-container col relative-position row no-wrap q-anchor--skip" style="margin-bottom: -1px;"><textarea class="q-field__native q-placeholder" style="overflow: hidden; min-height: 100px; height: 0px;" tabindex="0" rows="1" aria-label="Additional information" id="f_01010101-0101-4101-8101-010101010101" maxlength="500"></textarea>
                  <div class="q-field__label no-pointer-events absolute ellipsis">Additional information</div>
                  <!---->
                </div>
              </div>
              <div class="q-field__bottom row items-start q-field__bottom--animated">
                <transition-stub>
                  <div class="q-field__messages col"></div>
                </transition-stub>
                <!---->
              </div>
            </div>
            <div class="q-field__after q-field__marginal row no-wrap items-center"><i class="q-icon notranslate material-icons" aria-hidden="true" role="presentation">undo</i></div>
          </label><button class="q-btn q-btn-item non-selectable no-outline q-btn--outline q-btn--rectangle text-primary q-btn--actionable q-focusable q-hoverable absolute-bottom-right bg-white" style="font-size: 8px; bottom: -10px;" tabindex="0" type="button"><span class="q-focus-helper"></span><span class="q-btn__content text-center col items-center q-anchor--skip justify-center row"><span class="block">Preview</span>
            <!----></span>
          </button></div>
        <div class="v-popper v-popper--theme-mentionable v-popper--theme-dropdown popper v-popper--theme-mentionable v-popper--theme-dropdown popper" offset="6" style="position: absolute;">
          <div></div>
        </div>
      </div>
      <!--v-if-->
    </div>
    <!--v-if-->
    <div class="row justify-end q-gutter-sm q-mt-lg">
      <!--v-if--><button class="q-btn q-btn-item non-selectable no-outline q-btn--standard q-btn--rectangle disabled" tabindex="-1" type="button" disabled="" aria-disabled="true"><span class="q-focus-helper"></span><span class="q-btn__content text-center col items-center q-anchor--skip justify-center row">Reset</span></button><button class="q-btn q-btn-item non-selectable no-outline q-btn--standard q-btn--rectangle bg-red text-white q-btn--actionable q-focusable q-hoverable" tabindex="0" type="button"><span class="q-focus-helper"></span><span class="q-btn__content text-center col items-center q-anchor--skip justify-center row">Disable</span></button>
      <!--v-if--><button class="q-btn q-btn-item non-selectable no-outline q-btn--standard q-btn--rectangle bg-primary text-white disabled" tabindex="-1" type="submit" disabled="" aria-disabled="true"><span class="q-focus-helper"></span><span class="q-btn__content text-center col items-center q-anchor--skip justify-center row">Save changes</span>
        <transition-stub>
          <!---->
        </transition-stub>
      </button>
    </div>
  </form>
</div>
`;

exports[`Storyshots notifications/Notifications All Notifications 1`] = `
<div class="bg-white">
  <div class="full-width text-center generic-padding" style="display: none;"><svg class="q-spinner" fill="currentColor" width="40" height="40" viewBox="0 0 120 30" xmlns="http://www.w3.org/2000/svg">
      <circle cx="15" cy="15" r="15">
        <animate attributeName="r" from="15" to="15" begin="0s" dur="0.8s" values="15;9;15" calcMode="linear" repeatCount="indefinite"></animate>
        <animate attributeName="fill-opacity" from="1" to="1" begin="0s" dur="0.8s" values="1;.5;1" calcMode="linear" repeatCount="indefinite"></animate>
      </circle>
      <circle cx="60" cy="15" r="9" fill-opacity=".3">
        <animate attributeName="r" from="9" to="9" begin="0s" dur="0.8s" values="9;15;9" calcMode="linear" repeatCount="indefinite"></animate>
        <animate attributeName="fill-opacity" from=".5" to=".5" begin="0s" dur="0.8s" values=".5;1;.5" calcMode="linear" repeatCount="indefinite"></animate>
      </circle>
      <circle cx="105" cy="15" r="15">
        <animate attributeName="r" from="15" to="15" begin="0s" dur="0.8s" values="15;9;15" calcMode="linear" repeatCount="indefinite"></animate>
        <animate attributeName="fill-opacity" from="1" to="1" begin="0s" dur="0.8s" values="1;.5;1" calcMode="linear" repeatCount="indefinite"></animate>
      </circle>
    </svg></div>
  <div class="q-list">
    <!--v-if-->
    <div class="q-item q-item-type row no-wrap isUnread">
      <!--v-if-->
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label"><i class="q-icon fas fa-angle-double-up q-mx-xs vertical-baseline" aria-hidden="true" role="presentation"> </i> Georgia Hoeger gained editing permissions</div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="May 15, 2017, 6:42 AM" style="display: inline;">7 months ago</div> · enable Corporate bypassing Bedfordshire Tuna
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class="q-item q-item-type row no-wrap isUnread">
      <!--v-if-->
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label"><i class="q-icon fas fa-angle-double-up q-mx-xs vertical-baseline" aria-hidden="true" role="presentation"> </i> You gained editing permissions</div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="Sep 16, 2017, 6:48 PM" style="display: inline;">3 months ago</div> · enable Corporate bypassing Bedfordshire Tuna
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class="q-item q-item-type row no-wrap isUnread">
      <div class="q-item__section column q-item__section--side justify-center q-pr-sm">
        <div class="wrapper" style="width: 40px; height: 40px;"><a title="Connie Kunze">
            <div text="Connie Kunze" seed="1" class="randomArt fill"></div>
          </a></div>
      </div>
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label"><i class="q-icon fas fa-address-card q-mx-xs vertical-baseline" aria-hidden="true" role="presentation"> </i> applied to join your group</div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="Nov 9, 2017, 6:54 AM" style="display: inline;">about 2 months ago</div> · enable Corporate bypassing Bedfordshire Tuna
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class="q-item q-item-type row no-wrap isUnread">
      <!--v-if-->
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label"><i class="q-icon fas fa-check q-mx-xs vertical-baseline" aria-hidden="true" role="presentation"> </i> Your application was accepted!</div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="May 28, 2017, 12:33 PM" style="display: inline;">7 months ago</div> · enable Corporate bypassing Bedfordshire Tuna
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class="q-item q-item-type row no-wrap isUnread">
      <!--v-if-->
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label"><i class="q-icon fas fa-times q-mx-xs vertical-baseline" aria-hidden="true" role="presentation"> </i> Your application was declined</div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="Apr 15, 2017, 8:01 AM" style="display: inline;">8 months ago</div> · enable Corporate bypassing Bedfordshire Tuna
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class="q-item q-item-type row no-wrap isUnread">
      <!--v-if-->
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label">
          <!--v-if--> You can give feedback - Task on Monday 10:00 AM
        </div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="Mar 11, 2017, 8:02 AM" style="display: inline;">10 months ago</div> · enable Corporate bypassing Bedfordshire Tuna · leverage Automotive Jewelery Intelligent connecting
        </div>
      </div>
    </div>
    <div class="q-item q-item-type row no-wrap isUnread">
      <!--v-if-->
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label"><i class="q-icon fas fa-circle q-mx-xs vertical-baseline" aria-hidden="true" role="presentation"> </i> leverage Automotive Jewelery Intelligent connecting has been created</div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="Feb 19, 2017, 8:59 AM" style="display: inline;">10 months ago</div> · enable Corporate bypassing Bedfordshire Tuna
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class="q-item q-item-type row no-wrap isUnread">
      <div class="q-item__section column q-item__section--side justify-center q-pr-sm">
        <div class="wrapper" style="width: 40px; height: 40px;"><a title="Connie Kunze">
            <div text="Connie Kunze" seed="1" class="randomArt fill"></div>
          </a></div>
      </div>
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label"><i class="q-icon fas fa-user-plus q-mx-xs vertical-baseline" aria-hidden="true" role="presentation"> </i> Connie Kunze joined your group</div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="Sep 21, 2017, 7:27 AM" style="display: inline;">3 months ago</div> · enable Corporate bypassing Bedfordshire Tuna
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class="q-item q-item-type row no-wrap isUnread">
      <div class="q-item__section column q-item__section--side justify-center q-pr-sm">
        <div class="wrapper" style="width: 40px; height: 40px;"><a title="Connie Kunze">
            <div text="Connie Kunze" seed="1" class="randomArt fill"></div>
          </a></div>
      </div>
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label"><i class="q-icon fas fa-user-plus q-mx-xs vertical-baseline" aria-hidden="true" role="presentation"> </i> Connie Kunze accepted your invitation</div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="Nov 4, 2017, 9:24 AM" style="display: inline;">about 2 months ago</div> · enable Corporate bypassing Bedfordshire Tuna
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class="q-item q-item-type row no-wrap isUnread">
      <!--v-if-->
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label">
          <!--v-if--> Reminder: Task at 10:00 AM
        </div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="Feb 4, 2018, 8:20 AM" style="display: inline;">in about 1 month</div> · enable Corporate bypassing Bedfordshire Tuna · leverage Automotive Jewelery Intelligent connecting
        </div>
      </div>
    </div>
    <div class="q-item q-item-type row no-wrap isUnread">
      <!--v-if-->
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label"><i class="q-icon fas fa-times q-mx-xs vertical-baseline" aria-hidden="true" role="presentation"> </i> Task on Monday, December 25 at 10:00 AM was disabled</div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="Jun 20, 2017, 10:48 AM" style="display: inline;">6 months ago</div> · enable Corporate bypassing Bedfordshire Tuna · leverage Automotive Jewelery Intelligent connecting
        </div>
      </div>
    </div>
    <div class="q-item q-item-type row no-wrap isUnread">
      <!--v-if-->
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label"><i class="q-icon fas fa-check q-mx-xs vertical-baseline" aria-hidden="true" role="presentation"> </i> Task on Monday, December 25 at 10:00 AM was enabled again</div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="May 13, 2017, 9:15 AM" style="display: inline;">8 months ago</div> · enable Corporate bypassing Bedfordshire Tuna · leverage Automotive Jewelery Intelligent connecting
        </div>
      </div>
    </div>
    <div class="q-item q-item-type row no-wrap isUnread">
      <!--v-if-->
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label"><i class="q-icon far fa-clock q-mx-xs vertical-baseline" aria-hidden="true" role="presentation"> </i> Task was moved to Monday, December 25 at 10:00 AM</div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="Aug 20, 2017, 12:13 PM" style="display: inline;">4 months ago</div> · enable Corporate bypassing Bedfordshire Tuna · leverage Automotive Jewelery Intelligent connecting
        </div>
      </div>
    </div>
    <div class="q-item q-item-type row no-wrap isUnread">
      <div class="q-item__section column q-item__section--side justify-center q-pr-sm">
        <div class="wrapper" style="width: 40px; height: 40px;"><a title="Connie Kunze">
            <div text="Connie Kunze" seed="1" class="randomArt fill"></div>
          </a></div>
      </div>
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label"><i class="q-icon far fa-frown-open q-mx-xs vertical-baseline" aria-hidden="true" role="presentation"> </i> Join the discussion about a conflict with Connie Kunze</div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="Jul 11, 2017, 3:52 AM" style="display: inline;">6 months ago</div> · enable Corporate bypassing Bedfordshire Tuna
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class="q-item q-item-type row no-wrap isUnread">
      <!--v-if-->
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label"><i class="q-icon far fa-frown-open q-mx-xs vertical-baseline" aria-hidden="true" role="presentation"> </i> A conflict resolution process was started with you</div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="Jun 22, 2017, 5:42 PM" style="display: inline;">6 months ago</div> · enable Corporate bypassing Bedfordshire Tuna
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class="q-item q-item-type row no-wrap isUnread">
      <div class="q-item__section column q-item__section--side justify-center q-pr-sm">
        <div class="wrapper" style="width: 40px; height: 40px;"><a title="Connie Kunze">
            <div text="Connie Kunze" seed="1" class="randomArt fill"></div>
          </a></div>
      </div>
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label"><i class="q-icon far fa-frown-open q-mx-xs vertical-baseline" aria-hidden="true" role="presentation"> </i> The conflict resolution process with Connie Kunze continues</div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="May 24, 2017, 9:28 PM" style="display: inline;">7 months ago</div> · enable Corporate bypassing Bedfordshire Tuna
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class="q-item q-item-type row no-wrap isUnread">
      <!--v-if-->
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label"><i class="q-icon far fa-frown-open q-mx-xs vertical-baseline" aria-hidden="true" role="presentation"> </i> The conflict resolution process with you continues</div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="Aug 6, 2017, 1:12 AM" style="display: inline;">5 months ago</div> · enable Corporate bypassing Bedfordshire Tuna
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class="q-item q-item-type row no-wrap isUnread">
      <div class="q-item__section column q-item__section--side justify-center q-pr-sm">
        <div class="wrapper" style="width: 40px; height: 40px;"><a title="Connie Kunze">
            <div text="Connie Kunze" seed="1" class="randomArt fill"></div>
          </a></div>
      </div>
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label"><i class="q-icon far fa-frown-open q-mx-xs vertical-baseline" aria-hidden="true" role="presentation"> </i> The conflict resolution process with Connie Kunze was decided</div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="May 11, 2017, 5:48 AM" style="display: inline;">8 months ago</div> · enable Corporate bypassing Bedfordshire Tuna
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class="q-item q-item-type row no-wrap isUnread">
      <!--v-if-->
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label"><i class="q-icon far fa-frown-open q-mx-xs vertical-baseline" aria-hidden="true" role="presentation"> </i> The conflict resolution process with you was decided</div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="Nov 26, 2017, 2:16 PM" style="display: inline;">28 days ago</div> · enable Corporate bypassing Bedfordshire Tuna
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class="q-item q-item-type row no-wrap isUnread">
      <!--v-if-->
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label"><i class="q-icon fas fa-times q-mx-xs vertical-baseline" aria-hidden="true" role="presentation"> </i> You were removed from the group</div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="Nov 24, 2017, 2:34 PM" style="display: inline;">30 days ago</div> · enable Corporate bypassing Bedfordshire Tuna
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class="q-item q-item-type row no-wrap isUnread">
      <!--v-if-->
      <div class="q-item__section column q-item__section--main justify-center">
        <div style="overflow: hidden; display: -webkit-box;" class="q-item__label"><i class="q-icon far fa-clock q-mx-xs vertical-baseline" aria-hidden="true" role="presentation"> </i> Voting ends soon, don't forget to participate!</div>
        <div class="q-item__label q-item__label--caption text-caption">
          <div title="Apr 7, 2018, 8:36 PM" style="display: inline;">in 3 months</div> · enable Corporate bypassing Bedfordshire Tuna
          <!--v-if-->
        </div>
      </div>
    </div>
    <!--v-if-->
    <!--v-if-->
  </div>
</div>
`;
